frontend/src/modules/studySamples/services.ts

```diff
@@ -163,64 +129,61 @@ export async function buildStudySamplesFromWorkflow(
 
 	// Get the user ID's for the processes
 	const processesUserIDs = processes.map(process => process.created_by)
-  // Get the user ID's for the step history without processes
-  const stepHistoryUserIDs = completedSamplesByStudy.filter((completed) => isNullish(completed.process_measurement)).map(completed => completed.created_by)
-  const userIDs = processesUserIDs.concat(stepHistoryUserIDs)
+
+	// Get the user ID's for the step history without processes
+	const userIDs = processesUserIDs
 	const users = await fetchUsers(userIDs)
 	const usersByID = createItemsByID(users)
 
-	completedSamplesByStudy.forEach(stepHistory => {
-		const step = stepMap.get(stepHistory.step_order)
-		if (step) {
-			const processMeasurement = processMeasurementsByID[stepHistory.process_measurement]
-			const process = processMeasurement ? processesByID[processMeasurement.process] : undefined
-			const user = process && process.created_by ? usersByID[process.created_by] : stepHistory.created_by ? usersByID[stepHistory.created_by] : undefined
-			
-			const completedSample : CompletedStudySample = {
-				id: stepHistory.id,
-				sampleID: stepHistory.sample,
-				generatedSampleID: processMeasurement?.child_sample,
-				processID: processMeasurement?.process,
-				processMeasurementID: stepHistory.process_measurement,
-				executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(stepHistory.created_at),
-				executedBy: user?.username,
-				comment: processMeasurement?.comment,
-				removedFromWorkflow: stepHistory.workflow_action === 'DEQUEUE_SAMPLE'
-			}
-			step.completed.push(completedSample)
-		}
+	const completedStudySamples = completedSamples.map((completedSample) => {
+		const processMeasurement = processMeasurementsByID[completedSample.process_measurement]
+		const process = processMeasurement ? processesByID[processMeasurement.process] : undefined
+		const user = process && process.created_by ? usersByID[process.created_by] : completedSample.created_by ? usersByID[completedSample.created_by] : undefined
+
+		return {
+			id: completedSample.id,
+			sampleID: completedSample.sample,
+			generatedSampleID: processMeasurement?.child_sample,
+			processID: processMeasurement?.process,
+			processMeasurementID: completedSample.process_measurement,
+			executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(completedSample.created_at),
+			executedBy: user?.username,
+			comment: processMeasurement?.comment,
+		} as CompletedStudySample
 	})
 
-	// Return the steps in the step order
-	const steps = Array.from(stepMap.values()).sort((a, b) => a.stepOrder - b.stepOrder)
-
 	return {
-		steps
+		result: completedStudySamples,
+		count: totalCount
 	}
 }
 
-export async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: FMSId) {
-
+async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: number, offset = 0, limit = 10) {
```

Comment:
limit uses a duplication of a harcoded value ... use reference to the same constant as page_size ...